discovery_child_development/pipeline/03_prepare_training_data.py

```diff
@@ -0,0 +1,91 @@
+"""
+Prepare labelled data for training a classifier
+"""
+
+import pandas as pd
+import os
+from sklearn.model_selection import train_test_split
+
+from nesta_ds_utils.loading_saving import S3
+
+from discovery_child_development import PROJECT_DIR, logging
+from discovery_child_development.getters import taxonomy, openalex
+from discovery_child_development.utils.io import import_config
+
+S3_BUCKET = os.environ["S3_BUCKET"]
+
+PARAMS = import_config("config.yaml")
+
+CONCEPT_IDS = "|".join(PARAMS["openalex_concepts"])
+YEARS = [str(y) for y in PARAMS["openalex_years"]]
+YEARS = "-".join(YEARS)
+
+OUT_PATH = "data/openAlex/processed/"
+
+# needed for train-test split
+SEED = PARAMS["seed"]
+
+if __name__ == "__main__":
+    taxonomy_data = taxonomy.get_taxonomy()
+
+    # check that there are no duplicate concepts - this should return 0
+    len(taxonomy_data) - len(taxonomy_data["display_name"].unique())
+
+    # check that the number of unique names and the number of unique ids are the same
+    len(taxonomy_data["display_name"].unique()) == len(
+        taxonomy_data["concept_id"].unique()
+    )
+
+    # Get the IDs of concepts that we will use to filter the OpenAlex data
+    taxonomy_concept_ids = taxonomy_data["concept_id"].unique()
+
+    openalex_data = openalex.get_abstracts()
+
+    openalex_concepts = openalex.get_concepts_metadata()
+
+    openalex_concepts_subset = openalex_concepts[
+        openalex_concepts["concept_id"].isin(taxonomy_concept_ids)
+    ].copy()
+    logging.info(f"N rows lost: {len(openalex_concepts)-len(openalex_concepts_subset)}")
+
+    # merge taxonomy
+    openalex_concepts_subset = pd.merge(
+        openalex_concepts_subset,
+        taxonomy_data[["sub_category", "concept_id"]],
+        how="left",
+        on="concept_id",
+    )
+
+    # Check whether any works have been lost because they were not tagged with any concepts from the taxonomy
+    len(openalex_concepts["openalex_id"].unique()) - len(
+        openalex_concepts_subset["openalex_id"].unique()
+    )
```

Comment:
I guess this one is not needed here anymore?

Reply:
Good spot! I have incorporated this into a `logging.info()` statement.